Public/Src/Utilities/Utilities/BuildXLReaderExtensions.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace BuildXL.Utilities
+{
+    /// <summary>
+    /// Extensions for <see cref="BuildXLReader"/>.
+    /// </summary>
+    public static class BuildXLReaderExtensions

```

Comment:
>BuildXLReaderExtensions [](start = 24, length = 23)

Is this needed? Can't you just add methods to the BuildXLReader/Writer directly.

Reply:
I'm thinking that all the derivative methods should go to the extension. Or, do you have any concern?

---
In reply to: [277360478](https://github.com/Microsoft/BuildXL/pull/159#discussion_r277360478) [](ancestors = 277360478)